src/nspawn/nspawn-util.c

```diff
@@ -23,6 +23,7 @@ int systemd_installation_has_version(const char *root, const char *minimal_versi
                         */
                        "/lib/systemd/libsystemd-shared-*.so",
                        "/lib64/systemd/libsystemd-shared-*.so",
+                       "/usr/lib/*/systemd/libsystemd-shared-*.so",
```

Comment:
Hmm, can we make this glob a bit tighter? There can be hundreds of directories that match `/usr/lib/*/` on a big installation, and resolving this glob can be a tad expensive. (E.g. on my machine, with caches cleared, it can be ~0.5 s). Maybe `/usr/lib/*-linux-*/libsystemd-shared-*.so` ?

Reply:
Hm, this is on my old, messy laptop, with thousands of packets installed:
```
 $ find  /usr/lib/ -maxdepth 1 -type d | wc -l
150
$ time ls /usr/lib/*/systemd/libsystemd-shared-*.so
/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so

real	0m0,007s
user	0m0,000s
sys	0m0,007s
```

I actually thought about this issue but figured it's not much of an issue even if a tad inefficient, as this function is called so rarely. Especially, since you would call it inside an nspawn container, which is typically much more minimal and doesn't have 100s of directories.
We also exit early, so if you have a Fedora container, you shouldn't get the "expensive" glob, right?
Sorting this glob to the end of the list might be a good idea.
I was also considering including `ROOTPKGLIBDIR "/libsystemd-shared-*.so"` in that list. So as long as the architecture matches (which should be the predominant case), we'd try an exact match first.

Reply:
On the same system with the 150 directories
```
$ time ./build-deb/test-nspawn-util 
/* test_systemd_installation_has_version */
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (OK).
Current installation has systemd >= 0: yes
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (OK).
Current installation has systemd >= 231: yes
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (OK).
Current installation has systemd >= 251: yes
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (too old).
Current installation has systemd >= 999: no

real	0m0,011s
user	0m0,001s
sys	0m0,011s
```

Reply:
Using `/usr/lib/*-linux-*/systemd/libsystemd-shared-*.so`:

```
$ time ./build-deb/test-nspawn-util 
/* test_systemd_installation_has_version */
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (OK).
Current installation has systemd >= 0: yes
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (OK).
Current installation has systemd >= 231: yes
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (OK).
Current installation has systemd >= 251: yes
Found libsystemd shared at "/usr/lib/x86_64-linux-gnu/systemd/libsystemd-shared-251.so", version 251 (too old).
Current installation has systemd >= 999: no

real	0m0,009s
user	0m0,005s
sys	0m0,004s
```
A very slight improvement

Reply:
Given that using an explicit `ROOTPKGLIBDIR` didn't bring any improvements I went with your suggestion and used `*-linux-*`. I've updated the comment in the source code as well.